tools/federation/src/main/java/org/eclipse/rdf4j/federated/QueryManager.java

```diff
@@ -52,30 +57,43 @@
 
 	private static final Logger log = LoggerFactory.getLogger(QueryManager.class);
 
-	// singleton behavior: initialized in constructor of FederationManager
-	protected static QueryManager instance = null;
-
-	protected static QueryManager getInstance() {
-		return instance;
-	}
-
 	protected final FederationManager federationManager;
 	protected final Repository repo;
-	protected final RepositoryConnection conn;
+	protected RepositoryConnection conn;
 	protected final AtomicBigInteger nextQueryID;
 	protected Set<QueryInfo> runningQueries = new ConcurrentSkipListSet<QueryInfo>();
-	protected HashMap<String, String> prefixDeclarations = new HashMap<String, String>();
+	protected Map<String, String> prefixDeclarations = new HashMap<String, String>();
 
-	protected QueryManager(FederationManager federationManager, Repository repo) {
+	public QueryManager(FederationManager federationManager, Repository repo) {
 		this.federationManager = federationManager;
 		this.repo = repo;
+
+		BigInteger lastQueryId = new BigInteger("0");
+		this.nextQueryID = new AtomicBigInteger(lastQueryId);
+	}
+
+	public void initialize() {
```

Comment:
I realize this one is not overriding any existing methods, but given #552 perhaps we should rename this to `init` as well?

Reply:
Done